### DISCUSSION TITLE
Feedback for Personality Quiz Activity

### DISCUSSION THREAD

**Discussion by jcollard** [CaptainCoderOrg/IntroToCSharpSite#166] [Page Feedback]
Feedback for If Statements lesson: https://csharp-develop.captaincoder.org/activities/personality-quiz

**Comment by cloudthepain** [comment]
Do you think there is a better way to write these? I can't put my finger on it but something about the way this reads is off to me.
--Sum the scores to get a total possible result (or calculate it by hand).
--For each score, calculate the percentage score. Hint: You'll need to use a double.

Honestly, it could just be me. That was the only issue that I could find.
@jcollard

**Reply by cloudthepain** [reply]
Also, did you want to add a Replit at the end of the section to show them how it could possibly look and work?

**Reply by jcollard** [reply]
`Also, did you want to add a Replit at the end of the section to show them how it could possibly look and work?`

Maybe. I don't want a student to be able to copy and paste an entire project (or a large portion of one). I think I may end up writing a browser based quiz at some point to create a demonstration. Unfortunately, Replit removed the feature where you could run a console program without sharing the source code.

**Reply by jcollard** [reply]
> Do you think there is a better way to write these? I can't put my finger on it but something about the way this reads is off to me. --Sum the scores to get a total possible result (or calculate it by hand). --For each score, calculate the percentage score. Hint: You'll need to use a double.
> 
> Honestly, it could just be me. That was the only issue that I could find. @jcollard

Yeah, I agree they could use some rewording. Maybe:

```
Create a variable to store the total number of points.

Note: The total number of points should be the sum of all the scores.
```

And:

```
Using a double, calculate the score for each outcome by dividing the score by the total

Note: This will result in a number that is between 0 and 1. To convert the result to a percentage, you will need to multiply by 100.
```

Thoughts?

**Comment by Amun5** [comment]
help i cant figure out what to do for the last part
int fireScore = 0;
int waterScore = 0;
int earthScore = 0;
int airScore = 0;
Console.WriteLine("Welcome to the elemntal test");

System.Console.WriteLine("You just got home from work what would you chose to do ");
System.Console.WriteLine("1: Play some video games");
System.Console.WriteLine("2: Go to the gym");
System.Console.WriteLine("3: Cook some food;");
System.Console.WriteLine("4: Take a nap");

System.Console.WriteLine("Select one of the 4 choies");
string userInput = Console.ReadLine();

if(userInput == "1"){    
    fireScore = fireScore + 1;
}
else if (userInput == "1"){
    earthScore = earthScore + 1;
}
else if (userInput == "3"){
waterScore = waterScore + 1;
}
else if(userInput == "4"){
airScore =  airScore + 1;
}
else{
    System.Console.WriteLine("Invalid!!!");
}

System.Console.WriteLine("Flying or super speed");
System.Console.WriteLine("1:");
System.Console.WriteLine("2:");
string userInput1 = Console.ReadLine();

if (userInput1 == "1"){
    waterScore = waterScore + 1;
    airScore = airScore + 1;   
}
else if (userInput1 == "2"){
    fireScore = fireScore + 1;
    earthScore = earthScore + 1;
}
else {
    System.Console.WriteLine("INVALID!!!");
}

System.Console.WriteLine("pick a animal ");
System.Console.WriteLine("1: Wolf");
System.Console.WriteLine("2: Tiger");
System.Console.WriteLine("3: Snake");
System.Console.WriteLine("4: bear");
string userInput2 = Console.ReadLine();

if (userInput2 == "1"){
    airScore = airScore + 1;
}
else if (userInput2 == "2"){
    fireScore = fireScore + 1;
}
else if (userInput2 == "3"){
    airScore = airScore + 1;
}
else if (userInput2 == "4"){
    waterScore = waterScore + 1;
}
else{
    System.Console.WriteLine("Invalid");
}

System.Console.WriteLine("Pick one");
System.Console.WriteLine("1: Copper");
System.Console.WriteLine("2: Gold");
System.Console.WriteLine("3: Silver");
System.Console.WriteLine("4: Emreld");
string userInput3 = Console.ReadLine();

if (userInput3 == "1"){
    airScore = airScore +1;
}
else if (userInput3 == "2"){
    fireScore = fireScore + 1;
}
else if (userInput3 == "3"){
    waterScore = waterScore + 1;
}
else if (userInput3 == "4"){
    earthScore = earthScore + 1;
}
else {
    System.Console.WriteLine("Invalid");
}

System.Console.WriteLine("night or day ");
System.Console.WriteLine("1: Night");
System.Console.WriteLine("2: Day");
string userInput4 = Console.ReadLine();


if(userInput4 == "1"){
waterScore = waterScore + 1;
airScore = airScore + 1;
}
else if (userInput4 == "2"){
    fireScore = fireScore + 1;
    earthScore = earthScore + 1;
}
else{
    System.Console.WriteLine("Invalid");
}

System.Console.WriteLine("You have completed the quiz! Results");

double totalScore = fireScore + waterScore + earthScore + airScore;

double TotalfireScore = (fireScore/totalScore)* 100; 
double TotalwaterScore = (waterScore/totalScore* 100);
double TotalearthScore = (earthScore/totalScore)* 100;
double TotalairScore = (airScore/totalScore)* 100 ;


System.Console.WriteLine("Fire: " + TotalfireScore + "%");
System.Console.WriteLine("Water: " + TotalwaterScore + "%");
System.Console.WriteLine("Earth: " + TotalearthScore + "%");
System.Console.WriteLine("Air: " + TotalairScore +"%");


double maxScore = Math.Max(0,0);
maxScore = Math.Max(TotalfireScore, fireScore);
maxScore = Math.Max(TotalwaterScore, waterScore);
maxScore = Math.Max(TotalearthScore, earthScore);
maxScore = Math.Max(TotalairScore, airScore);
System.Console.WriteLine(maxScore);

if(maxScore == TotalfireScore){
System.Console.WriteLine("You are a fire");
}

else if(maxScore == TotalwaterScore){
System.Console.WriteLine("You are a water");
}

else if(maxScore == TotalearthScore){
System.Console.WriteLine("You are a earth");
}

else if(maxScore == TotalairScore){
System.Console.WriteLine("You are a air ");
}

else{
    System.Console.WriteLine("IINVALID");
}

**Reply by jcollard** [reply]
This looks good! When you say the last part? What are you referring to? Also, feel free to hop on the discord to chat for more help :)

https://discord.gg/hEKtK2ySFC

**Comment by Amun5** [comment]
when i run the code i keep getting invalid so i don't know what part i messed up on

**Reply by jcollard** [reply]
Interesting. It works when I run it. What inputs do you provide?

**Reply by Amun5** [reply]
or it will print out one of the options that the user got 0% on as the final answer

**Reply by Amun5** [reply]
run it and pick 1, 2,2,2,2 that should print out the user is fire but it gives me water

**Reply by jcollard** [reply]
Ah, I found the bug:

```csharp
double maxScore = Math.Max(0, 0);
maxScore = Math.Max(TotalfireScore, maxScore); // <-- You need to take the max of the maxScore
maxScore = Math.Max(TotalwaterScore, maxScore);
maxScore = Math.Max(TotalearthScore, maxScore);
maxScore = Math.Max(TotalairScore, maxScore);
System.Console.WriteLine(maxScore);
```

**Reply by Amun5** [reply]
thx a lot been stuck on it all day, it working now but i cant say i understand why that fixed it

**Reply by jcollard** [reply]
Previously you had:

```csharp
double maxScore = Math.Max(0, 0);
maxScore = Math.Max(TotalfireScore, fireScore);
maxScore = Math.Max(TotalwaterScore, waterScore);
maxScore = Math.Max(TotalearthScore, earthScore);
maxScore = Math.Max(TotalairScore, airScore);
System.Console.WriteLine(maxScore);
```

In the above code `maxScore` will always be the max of `TotalairScore` and `airScore` this is because the final assignment is:

```csharp
maxScore = Math.Max(TotalairScore, airScore);
```

All of the previous assignments are essentially ignored. When you use `maxScore` in each of the calls, you are always taking the largest of `maxScore` and one of the other scores.

I hope that helps!
